src/particles/pusher/BeamParticleAdvance.cpp

```diff
@@ -18,8 +18,8 @@ AdvanceBeamParticlesSlice (BeamParticleContainer& beam, Fields& fields, amrex::G
     const PhysConst phys_const = get_phys_const();
 
     const bool do_z_push = beam.m_do_z_push;
-
-    const amrex::Real dt = Hipace::m_dt;
+    const int n_subcycles = beam.m_n_subcycles;
+    const amrex::Real dt = Hipace::m_dt / n_subcycles;
```

Comment:
Why not?
```suggestion
    const amrex::Real dt = Hipace::m_dt / beam.m_n_subcycles;
```

Reply:
Ah, this does not work because I need the ` n_subcycles` in the GPU loop as it must not be a member variable there.